frontend/src/pages/HomePage.tsx

```diff
@@ -1,9 +1,83 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useState, useEffect } from 'react';
 import ApartmentCard from '../components/Home/ApartmentCard';
-import { Box, Container, Grid, Typography } from '@material-ui/core';
+import { Box, Container, Grid, Typography, Link } from '@material-ui/core';
 import Autocomplete from '../components/Home/Autocomplete';
+import { Apartment, LandlordWithId, Review } from '../../../common/types/db-types';
+import { Link as RouterLink } from 'react-router-dom';
+import get from '../utils/get';
 
 const HomePage = (): ReactElement => {
+  const [homeData, setHomedata] = useState<any>([]);
+  const [buildingData, setBuildingData] = useState<Apartment[]>([]);
+  const [landlordData, setLandlordData] = useState<LandlordWithId[]>([]);
+  const [reviewsBuilding1, setReviewsBuilding1] = useState<Review[]>([]);
+  const [reviewsBuilding2, setReviewsBuilding2] = useState<Review[]>([]);
+  const [reviewsBuilding3, setReviewsBuilding3] = useState<Review[]>([]);
+  const [allReviews, setAllReviews] = useState<Review[][]>([]);
+  const [loaded, setLoaded] = useState(false);
+
+  useEffect(() => {
+    get<any>(`/homepageData`, setHomedata, undefined);
+  }, []);
+
+  useEffect(() => {
+    setBuildingData(homeData.buildings);
+  }, [homeData]);
+
+  useEffect(() => {
+    setLandlordData(homeData.landlords);
+  }, [homeData]);
+
+  useEffect(() => {
+    if (buildingData && buildingData.length > 0) {
+      get<Review>(
+        `/reviews/landlordId/${buildingData[0].landlordId}`,
+        setReviewsBuilding1,
+        undefined
+      );
+    }
+  }, [buildingData]);
+
+  useEffect(() => {
+    if (buildingData && buildingData.length > 1) {
+      get<Review>(
+        `/reviews/landlordId/${buildingData[1].landlordId}`,
+        setReviewsBuilding2,
+        undefined
+      );
+    }
+  }, [buildingData]);
+
+  useEffect(() => {
+    if (buildingData && buildingData.length > 2) {
+      get<Review>(
+        `/reviews/landlordId/${buildingData[2].landlordId}`,
+        setReviewsBuilding3,
+        undefined
+      );
+    }
+  }, [buildingData]);
```

Comment:
lots of repeated code here that can be deduplicated by just making the building states a single state as described above